### DISCUSSION TITLE
snapshotProcessor modifies the type it wraps

### DISCUSSION THREAD

**Discussion by scytacki** [mobxjs/mobx-state-tree#1896] [Q&A]
Is it expected that `types.snapshotProcessor` will modify the type that it is wrapping?

I found the following test passes:
```
    const Todo1 = types.model({ text: types.maybe(types.string) });
    const Todo2 = types.snapshotProcessor(Todo1, {
      preProcessor(sn) {
        return {text: "todo2 text"};
      }
    });
    const todo1a = Todo1.create();
    expect(todo1a.text).toBeUndefined();

    const todo2 =Todo2.create();
    expect(todo2.text).toBe("todo2 text");

    const todo1b = Todo1.create();
    // Is this really expected??
    expect(todo1b.text).toBe("todo2 text");
```

Note that the modification only happens after `Todo2.create()` is called. 

I tracked this down to `SnapshotProcessor#_fixNode` which is called on instantiate, and it calls `proxyNodeTypeMethods(node.type, this, "create")`. 

If this is unexpected, I can make an issue for it.

**Comment by EmilTholin** [comment]
Hi @scytacki!

Hmm, nice find. That looks like a bug, for sure.

**Reply by scytacki** [reply]
Hello @EmilTholin,

Thanks for responding.
Here is the issue I created about it: https://github.com/mobxjs/mobx-state-tree/issues/1897
